driver/src/main/java/org/neo4j/driver/v1/CypherTypes.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.v1;
+
+import org.neo4j.driver.v1.internal.types.TypeConstructor;
+import org.neo4j.driver.v1.internal.types.TypeRepresentation;
+import org.neo4j.driver.v1.internal.value.ValueAdapter;
+
+import static org.neo4j.driver.v1.internal.types.TypeConstructor.*;
+
+/**
+ * Utility class for determining and working with the Cypher types of values
+ *
+ * @see Value
+ * @see CypherType
+ * @see CoarseCypherType
+ */
+public final class CypherTypes
+{
+    private CypherTypes() {
+        throw new UnsupportedOperationException(  );
+    }
+
+    /**
+     * Determine the {@link CoarseCypherType} of a given {@link Value}
+     *
+     * @param value the value
+     * @return the smallest coarse Cypher type of the value
+     */
+    public static CoarseCypherType typeOf( Value value )
+    {
+        if ( value == null )
+        {
+            return NULL;
+        }
+
+        TypeConstructor constructor = ((ValueAdapter) value).typeConstructor();
```

Comment:
Regarding why: null is Java null and hence can't have a method. Also doing it this way keeps it completely orthogonal and replaceable with a different type system.